eth/catalyst/api.go

```diff
@@ -430,11 +424,18 @@ func (api *ConsensusAPI) GetPayloadV1(payloadID engine.PayloadID) (*engine.Execu
 
 // GetPayloadV2 returns a cached payload by id.
 func (api *ConsensusAPI) GetPayloadV2(payloadID engine.PayloadID) (*engine.ExecutionPayloadEnvelope, error) {
+	version := payloadID.Version()
+	if version != engine.PayloadV1 && version != engine.PayloadV2 {
+		return nil, engine.UnsupportedFork
+	}
```

Comment:
Wouldn't this be neat? 
```suggestion
	if !payloadID.Is(engine.PayloadV1, engine.PayloadV2) {
		return nil, engine.UnsupportedFork
	}
```

We can have it if you add something like this
```golang
func (b PayloadID) Is(versions ...PayloadVersion) bool {
        for _, v := range versions{
             if v == PayloadVersion(b[0]) { return true }
        }
	return false
}
```

Reply:
added in 684e3e